rfc8252_http_server/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+            }
+            .render()
+            .unwrap(),
+        ))
+}
+
+async fn threaded_error(
+    oidc_params_sender: mpsc::Sender<StdResult<OidcResponseParams, String>>,
+    error: &str,
+    error_description: &str,
+) -> Result<HttpResponse> {
+    let _ = oidc_params_sender.send(Err(format!("{}: {}", error, error_description)));
+    Ok(HttpResponse::build(http::StatusCode::OK)
+        .content_type("text/html; charset=utf-8")
+        .body(
+            OIDCErrorPage {
+                product_docs_link: "https://www.mongodb.com/docs/atlas/data-federation/query/sql/drivers/odbc/connect",
+                product_docs_name: "Atlas SQL ODBC Driver",
+                error,
+                error_uri: "https://www.mongodb.com/docs/atlas/data-federation/query/sql/drivers/odbc/connect/bad_oidc_login",
+                error_description,
+            }
+            .render()
+            .unwrap(),
+        ))
+}
+
+async fn tokio_error(
+    oidc_params_sender: tokio_mpsc::Sender<StdResult<OidcResponseParams, String>>,
+    error: &str,
+    error_description: &str,
+) -> Result<HttpResponse> {
+    let _ = oidc_params_sender
+        .send(Err(format!("{}: {}", error, error_description)))
+        .await;
+    Ok(HttpResponse::build(http::StatusCode::OK)
+        .content_type("text/html; charset=utf-8")
+        .body(
+            OIDCErrorPage {
+                product_docs_link: "https://www.mongodb.com/docs/atlas/data-federation/query/sql/drivers/odbc/connect",
+                product_docs_name: "Atlas SQL ODBC Driver",
+                error,
+                error_uri: "https://www.mongodb.com/docs/atlas/data-federation/query/sql/drivers/odbc/connect/bad_oidc_login",
+                error_description,
+            }
+            .render()
+            .unwrap(),
+        ))
+}
+
+async fn not_found() -> Result<HttpResponse> {
+    Ok(HttpResponse::build(http::StatusCode::OK)
+        .content_type("text/html; charset=utf-8")
+        .body(
+            OIDCNotFoundPage {
+                product_docs_link: "https://www.mongodb.com/docs/atlas/data-federation/query/sql/drivers/odbc/connect",
+                product_docs_name: "Atlas SQL ODBC Driver",
+            }
+            .render()
+            .unwrap(),
+        ))
+}
+
+async fn threaded_run_app(
+    sender: mpsc::Sender<ServerHandle>,
+    oidc_params_sender: mpsc::Sender<StdResult<OidcResponseParams, String>>,
+) -> std::io::Result<()> {
+    // srv is server controller type, `dev::Server`
+    let server = HttpServer::new(move || {
+        let oidc_params_sender = oidc_params_sender.clone();
+        App::new()
+            // enable logger
+            .wrap(middleware::Logger::default())
+            .service(
+                web::resource("/callback")
+                    .to(move |r| threaded_callback(oidc_params_sender.clone(), r)),
+            )
+            .default_service(web::route().to(not_found))
+    })
+    .bind(("127.0.0.1", 9080))?
```

Comment:
It's possible port 9080 is already used as it is somewhat of a common webserver port.  
A couple suggestions to handle this:
1. Use a less commonly used port ie 51234
2. Let the OS assign the port by setting it to `0`.  It would be more difficult to determine it for testing though.  [Here](https://doc.rust-lang.org/std/net/struct.TcpListener.html#method.bind) it says: The port allocated can be queried via the TcpListener::[local_addr](https://doc.rust-lang.org/std/net/struct.TcpListener.html#method.local_addr) method.

Reply:
In the next PR I just used the same default as the javascript plugin, which is 27097. Let me look into local_addr. THe js plugin made it configurable, but we opted not to do that since that causes issues with jdbc/odbc interface

Reply:
I just pulled back my changes from SQL-1935, should cover the port for now.